deploy/docker/Dockerfile

```diff
@@ -1,28 +1,44 @@
 FROM alpine:3.5
 
-LABEL maintainer="adundovi@gmail.com"
-LABEL description="Prepares a docker image \
-of pump.io server, http://pump.io/."
+LABEL maintainer Andrej Dundovic <adundovi@gmail.com>
+LABEL description "Stream server that does \
+most of what people really want from a social network."
 
 ENV LC_ALL en_US.UTF-8
 
+ENV PUMPIO_VERSION 3.0.2
+
 RUN apk add --no-cache \
-  nodejs \
-  git \
-  python \
-  make \
-  g++ \
-  graphicsmagick
-RUN npm install -g databank-mongodb
-RUN npm install -g pump.io
-RUN adduser -D -g "" pumpio
+    curl \
+    nodejs \
+    python \
+    make \
+    g++ \
+    git \
+    graphicsmagick \
+ && adduser -D -g "" pumpio \
+ && mkdir -p /etc/ssl/private \
+ && npm install -g databank-mongodb@0.19.2 \
+ && mkdir -p /usr/src \
+ && curl -fSL https://github.com/pump-io/pump.io/archive/v$PUMPIO_VERSION.zip -o pumpio.zip \
```

Comment:
> For the former one we start a new repository which would be updated only when needed (ideally, rarely)

So I think this is the only controversial point here - the rest we're generally in agreement about. I'd like to again state that I don't see any compelling advantages to having a separate repository for the production Dockerfile, whereas I see advantages to keeping it in the same repo. What exactly is the problem with that? I know Dockerfiles are supposed to only rely on stuff in the directory they're in (where is that recommendation from, btw?) but I don't really think that's a compelling argument. Almost no one is going to download the Dockerfile on its own and run it; people are going to clone the repository. And we can document that that's how the build works for the tiny minority who won't do that.

Or if we really wanted to strictly adhere to the "only use things in the current directory" approach, couldn't we just have a `Dockerfile.dev` and `Dockerfile.prod` or something? Another reason I'm pushing to keep it in this repo is because it makes it easier to pull in new changes, especially if you patch things. It's the same reason I strongly recommend avoiding GitHub's tarball downloads and instead using a git clone (that is, if you really _must_ not use the npm package). It just makes things easier to keep track of.